src/frontend/js/hooks/useStepManager/index.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useMemo, useState } from 'react';
+import { Nullable } from 'types/utils';
+
+export interface Step {
+  icon?: string;
+  label?: string;
+  next: Nullable<string>;
+  onEnter?: Function;
+  onExit?: Function;
+}
+
+export interface Manifest {
+  start: string;
+  steps: {
+    [key: string]: Step;
```

Comment:
```suggestion
    [key: string]: Step;
```
start key is not necessarily included in step, is there's something we could do about this?

Reply:
Good idea! I should be able to do that.

Reply:
I found a workaround by providing a generic type to `Manifest` and pass down this one to `Step` type.